external-crates/move/crates/move-abstract-interpreter/src/absint.rs

```diff
@@ -1,228 +1,151 @@
-
-        while let Some(block_id) = next_block {
-            let block_invariant = match inv_map.get_mut(&block_id) {
-                Some(invariant) => invariant,
-                None => {
-                    // This can only happen when all predecessors have errors,
-                    // so skip the block and move on to the next one
-                    next_block = function_context.cfg().next_block(block_id);
-                    continue;
-                }
-            };
-
-            let pre_state = &block_invariant.pre;
-            // Note: this will stop analysis after the first error occurs, to avoid the risk of
-            // subsequent crashes
-            let post_state = self.execute_block(block_id, pre_state, function_context, meter)?;
-
-            let mut next_block_candidate = function_context.cfg().next_block(block_id);
-            // propagate postcondition of this block to successor blocks
-            for successor_block_id in function_context.cfg().successors(block_id) {
-                meter.add(Scope::Function, PER_SUCCESSOR_COST)?;
-                match inv_map.get_mut(successor_block_id) {
-                    Some(next_block_invariant) => {
-                        let join_result = {
-                            let old_pre = &mut next_block_invariant.pre;
-                            old_pre.join(&post_state, meter)
-                        }?;
-                        match join_result {
-                            JoinResult::Unchanged => {
-                                // Pre is the same after join. Reanalyzing this block would produce
-                                // the same post
-                            }
-                            JoinResult::Changed => {
-                                // If the cur->successor is a back edge, jump back to the beginning
-                                // of the loop, instead of the normal next block
-                                if function_context
-                                    .cfg()
-                                    .is_back_edge(block_id, *successor_block_id)
-                                {
-                                    meter.add(Scope::Function, PER_BACKEDGE_COST)?;
-                                    next_block_candidate = Some(*successor_block_id);
-                                    break;
-                                }
+/// Analyze procedure local@function_context starting from pre-state local@initial_state.
+pub fn analyze_function<A, CFG>(
+    interpreter: &mut A,
+    cfg: &CFG,
+    code: &<CFG as ControlFlowGraph>::Instructions,
+    initial_state: A::State,
+) -> Result<(), A::Error>
+where
+    A: AbstractInterpreter,
+    CFG: ControlFlowGraph<
+        BlockId = A::BlockId,
+        InstructionIndex = A::InstructionIndex,
+        Instruction = A::Instruction,
+    >,
+{
+    interpreter.start()?;
+    // meter.add(Scope::Function, ANALYZE_FUNCTION_BASE_COST)?;
```

Comment:
What is this comment for?

Reply:
Leftover to make sure I got them in external-crates/move/crates/move-bytecode-verifier/src/absint.rs
I'll remove